src/networking.c

```diff
@@ -1836,10 +1741,9 @@ void freeClient(client *c) {
     if (c->name) decrRefCount(c->name);
     if (c->lib_name) decrRefCount(c->lib_name);
     if (c->lib_ver) decrRefCount(c->lib_ver);
-    freeClientMultiState(c);
+    if (c->mstate) freeClientMultiState(c);
```

Comment:
In the implementation of `freeClientMultiState`, I see code like `if (!c->mstate) return;`. So we check twice if `c->mstate` exists. Is that intentional?

If we only want to check it in one place, which place is better? If we check at the call site, the we can change the check in the function to an assert. Or if we check in the implementation, we don't need the check here and can just call `freeClientMultiState(c);` unconditionally. Or is check twice better for branch prediction or something?

It's a similar situation for freeClientPubSubData, freeClientModuleData, blocking state and some other client sub-structs.

Which style do you prefer?

Reply:
Since this call occurs only once per connection, branch prediction is not critical. I will remove the external conditions.